torch/distributed/distributed_c10d.py

```diff
@@ -2797,3 +2791,124 @@ def new_group(ranks=None, timeout=default_pg_timeout, backend=None, pg_options=N
             pg._set_sequence_number_for_group()
 
     return pg
+
+
+def new_subgroups(
+    ranks_per_subgroup_list=None,
+    group=None,
+    timeout=default_pg_timeout,
+    backend=None,
+    pg_options=None,
+):
+    """
```

Comment:
Thanks for adding these docs! Can you also add a basic example into the docs on how the API can be used and an example of the cleanup process?